layers/+lang/go/funcs.el

```diff
@@ -11,7 +11,8 @@
 
 (defun spacemacs//go-set-tab-width ()
   "Set the tab width."
-  (setq-local tab-width go-tab-width))
+  (if go-tab-width
+      (setq-local tab-width go-tab-width)))
```

Comment:
When you do not have an else clause, it is a good idea to use `(when a b)` instead of `(if a b)`.  Besides stylistic preference, it is all too easy for someone to change your `(if a b)` to `(if a b c)` (if a is true, then evaluate and return b, else evaluate and return c) when the intention was `(if a (progn b c))` (if a is true, then evaluate b and then evaluate and return c).

Reply:
OK, updated, thank you ! 
Also thank you for the PR problem, I have revert my branch